src/shogun/evaluation/MachineEvaluation.cpp

```diff
@@ -121,5 +165,19 @@ CMachine* CMachineEvaluation::get_machine() const
 
 EEvaluationDirection CMachineEvaluation::get_evaluation_direction()
 {
-	return m_evaluation_criterion->get_evaluation_direction();
+    REQUIRE(m_list_evaluation_criterion->get_array_size()==1,"Multiple Metrics provided. Please use get_evaluation_directions()");
+    CEvaluation* temp_eval = (CEvaluation*) m_list_evaluation_criterion->get_element_safe(0);
+    return temp_eval->get_evaluation_direction();
+}
+
+CDynamicArray<EEvaluationDirection> CMachineEvaluation::get_evaluation_directions()
+{
+    CDynamicArray<EEvaluationDirection> list_evaluation_direction = *(new CDynamicArray<EEvaluationDirection>());
```

Comment:
Please dont do return by value for Shogun classes.
Rather use pointers, also make sure to increase the refcount, and also tell SWIG that this method returns a new object.
See Evaluation.i